File: [23.12]Design-Pattern/Java/src/behavioral/chainOfResponsibility/LoginHandler.java
package behavioral.chainOfResponsibility;

public abstract class LoginHandler implements Handler{

    Handler handler;

    @Override
    public void setNextHandler(Handler handler) {
        this.handler = handler;
    }

    @Override
    public void process(String authority) {
        try{
            this.handler.process(authority);
        }catch (Exception e){
            System.out.println("로그인 실패");
        }
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/chainOfResponsibility/LoginHandler.kt
package behavioral.chainOfResponsibility

abstract class LoginHandler(): Handler {
    lateinit var handler: Handler
    override fun setNextHandler(handler: Handler) {
        this.handler=handler
    }

    override fun process(authority: String) {
        try{
            this.handler.process(authority)
        }catch (exception: Exception){
            println("로그인 실패")
        }
    }
}